index.js

```diff
@@ -36,35 +36,18 @@ client.on('ready', () => {
 client.on('message', async (message) => {
   // prevent infinite bot looping
   if (message.author.bot) return;
-  // message must always begin with prefix
-  const normalMessage = message.content.toLowerCase();
-  if (normalMessage.indexOf(GLOBAL_PREFIX) !== 0) return;
-
-  // message is for botto-kun and not from enemy bot
-  // remove nasties from nice words
-  logger.info(`RAW MESSAGE: ${message.content}`);
-  const cleanMessage = normalMessage.replace(/[^a-z0-9 ]/g, '').slice(GLOBAL_PREFIX.length);
-  logger.info(`CLEAN MESSAGE: ${cleanMessage}`);
-
-  // split input text into arguments and capture the first as potential command
-  const args = cleanMessage.trim().split(/ +/g);
-  const command = args[0].toLowerCase();
 
-  // test potential command against command list
-  const isCommand = Object.keys(commands).includes(command);
-  // keep track of whether or not a response was sent
+  const action = transformMessage(message);
   let actionComplete = false;
```

Comment:
i really hate the action complete logic but I think it has to be this way otherwise certain bot commands that involve deleting and editing messages won't work

Reply:
yeah I think it's fine for now though